data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/model/LoadStatus.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.dynamodb.model;
+
+import java.util.Map;
+
+public class LoadStatus {
+
+    private static final String TOTAL_FILES = "totalFiles";
+    private static final String LOADED_FILES = "loadedFiles";
+    private static final String TOTAL_RECORDS = "totalRecords";
+    private static final String LOADED_RECORDS = "loadedRecords";
+
+    private int totalFiles;
+
+    private int loadedFiles;
+
+    private int totalRecords;
+
+    private int loadedRecords;
+
+    public LoadStatus(int totalFiles, int loadedFiles, int totalRecords, int loadedRecords) {
+        this.totalFiles = totalFiles;
+        this.loadedFiles = loadedFiles;
+        this.totalRecords = totalRecords;
+        this.loadedRecords = loadedRecords;
+    }
+
+    public int getTotalFiles() {
+        return totalFiles;
+    }
+
+    public void setTotalFiles(int totalFiles) {
```

Comment:
I think you can remove these setters. Then you can make the fields final which provides better encapsulation.